android/app/src/main/java/io/mosip/registration_client/MainActivity.java

```diff
@@ -147,94 +153,131 @@ public class MainActivity extends FlutterActivity {
             );
         } else {
             pendingIntent = PendingIntent.getService(
                     this,
                     0,  // Request code
-                    serviceIntent,
+                    intent,
                     PendingIntent.FLAG_UPDATE_CURRENT
             );
         }
-
-        // Get an instance of AlarmManager
         AlarmManager alarmManager = (AlarmManager) getSystemService(Context.ALARM_SERVICE);
-
-        // Set the alarm to trigger your PendingIntent after a certain interval
-        long delayMillis = 3600000;  // Example delay of 60 seconds
-        long triggerAtMillis = SystemClock.elapsedRealtime() + delayMillis;
-        alarmManager.setRepeating(
-                AlarmManager.ELAPSED_REALTIME_WAKEUP,
-                triggerAtMillis,
-                delayMillis,
-                pendingIntent
-        );
+        if (alarmManager != null) {
+            long alarmTime = getIntervalMillis(api);
+            long currentTime = System.currentTimeMillis();
+            long delay = alarmTime > currentTime ? alarmTime - currentTime : alarmTime - currentTime;
+            Log.d(getClass().getSimpleName(), String.valueOf(delay)+ " Next Execution");
+
+            if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.M) {
+                alarmManager.setExactAndAllowWhileIdle(AlarmManager.ELAPSED_REALTIME_WAKEUP, SystemClock.elapsedRealtime() + delay, pendingIntent);
+            } else if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.KITKAT) {
+                alarmManager.setExact(AlarmManager.ELAPSED_REALTIME_WAKEUP, SystemClock.elapsedRealtime() + delay, pendingIntent);
+            } else {
+                alarmManager.set(AlarmManager.ELAPSED_REALTIME_WAKEUP, SystemClock.elapsedRealtime() + delay, pendingIntent);
+            }
+        }
     }
 
-    private void fetchRegistrationPackets(Context context) {
+    private void syncRegistrationPackets(Context context) {
         if(NetworkUtils.isNetworkConnected(context)){
-            Log.d(getClass().getSimpleName(), "Fetching Packets in main activity");
-            List<Registration> registrationList = packetService.getAllRegistrations(1,5);
-            Log.e(getClass().getSimpleName(), "Registration : "+ registrationList);
-
-            registrationList = packetService.getRegistrationsByStatus("CREATED");
-            registrationList.forEach(value->{
+            Log.d(getClass().getSimpleName(), "Sync Packets in main activity");
+            Integer batchSize = getBatchSize();
+            List<Registration> registrationList = packetService.getRegistrationsByStatus(PacketClientStatus.APPROVED.name(), batchSize);
+            for (Registration value : registrationList) {
                 try {
-                    Log.d(getClass().getSimpleName(), "Syncing " + value.getPacketId());
+                    Log.e(getClass().getSimpleName(), "Syncing " + value.getPacketId());
```

Comment:
log.e should only be used where there's error being logged, use log.i or log.d for info or debug log

Reply:
Changed to log.d